services/121-service/src/user/user.service.spec.ts

```diff
@@ -0,0 +1,942 @@
+      expect(result).toMatchObject({
+        id: userRoleId,
+        role: 'test-role',
+        label: 'Test Role',
+        description: 'Test Description',
+        permissions: [],
+      });
+    });
+
+    it('should handle permission repository errors', async () => {
+      // Arrange
+      const updateData = { permissions: ['invalid-permission'] as any };
+      jest
+        .spyOn(permissionRepository, 'findOneByOrFail')
+        .mockRejectedValue(new Error('Permission not found'));
+
+      // Act & Assert
+      await expect(
+        service.updateUserRole(userRoleId, updateData),
+      ).rejects.toThrow('Permission not found');
+
+      expect(permissionRepository.findOneByOrFail).toHaveBeenCalledWith({
+        name: 'invalid-permission',
+      });
+    });
+  });
+
+  describe('deleteUserRole', () => {
+    const userRoleId = 1;
+    const mockExistingRole: Partial<UserRoleEntity> = {
+      id: userRoleId,
+      role: 'test-role',
+      label: 'Test Role',
+      description: 'Test Description',
+      permissions: [],
+    };
+
+    beforeEach(() => {
+      // Mock the private findRoleOrThrow method by spying on userRoleRepository.findOneBy
+      jest
+        .spyOn(userRoleRepository, 'findOneBy')
+        .mockResolvedValue(mockExistingRole as UserRoleEntity);
+      jest
+        .spyOn(userRoleRepository, 'remove')
+        .mockResolvedValue(mockExistingRole as UserRoleEntity);
+    });
+
+    it('should throw HttpException when role is not found', async () => {
+      // Arrange
+      jest.spyOn(userRoleRepository, 'findOneBy').mockResolvedValue(null);
+
+      // Act & Assert
+      await expect(service.deleteUserRole(userRoleId)).rejects.toThrow(
+        new HttpException('Role not found', HttpStatus.NOT_FOUND),
+      );
+
+      expect(userRoleRepository.remove).not.toHaveBeenCalled();
+    });
+
+    it('should successfully delete an existing role', async () => {
+      // Act
+      const result = await service.deleteUserRole(userRoleId);
+
+      // Assert
+      expect(userRoleRepository.remove).toHaveBeenCalledWith(mockExistingRole);
+      expect(result).toMatchObject({
+        id: userRoleId,
+        role: 'test-role',
+        label: 'Test Role',
+        description: 'Test Description',
+        permissions: [],
+      });
+    });
+
+    it('should handle repository remove errors', async () => {
+      // Arrange
+      const removeError = new Error('Database error during deletion');
+      jest.spyOn(userRoleRepository, 'remove').mockRejectedValue(removeError);
+
+      // Act & Assert
+      await expect(service.deleteUserRole(userRoleId)).rejects.toThrow(
+        'Database error during deletion',
+      );
+
+      expect(userRoleRepository.remove).toHaveBeenCalledWith(mockExistingRole);
+    });
+
+    it('should return the deleted role data in the correct format', async () => {
+      // Arrange
+      const mockDeletedRole = {
+        ...mockExistingRole,
+        id: userRoleId,
```

Comment:
```suggestion
```

This `id` already seems to be that same number... No need to 'override' it, I think?